src/kernel/bitcoinkernel.h

```diff
@@ -1568,6 +1641,92 @@ BITCOINKERNEL_API void btck_block_hash_destroy(btck_BlockHash* block_hash);
 
 ///@}
 
+/**
+ * @name Block Header
+ * Functions for working with block headers.
+*/
+///@{
+
+/**
+ * @brief Create a block header from serialized data.
+ *
+ * @param[in] raw_block_header      Non-null, serialized header data (80 bytes)
+ * @param[in] raw_block_header_len  Length of serialized header (must be 80)
+ * @return                          Block Header, or null on error. Must be destroyed with btck_block_header_destroy().
+ */
+BITCOINKERNEL_API btck_BlockHeader* BITCOINKERNEL_WARN_UNUSED_RESULT btck_block_header_create(
+    const void* raw_block_header, size_t raw_block_header_len) BITCOINKERNEL_ARG_NONNULL(1);
+
+/**
+ * @brief Copy a block header
+ *
+ * @param[in] header    Non-null header
+ * @return              Copied header. Must be destroyed with btck_block_header_destroy().
+ */
+BITCOINKERNEL_API btck_BlockHeader* BITCOINKERNEL_WARN_UNUSED_RESULT btck_block_header_copy(
+    const btck_BlockHeader* header) BITCOINKERNEL_ARG_NONNULL(1);
+
+/**
+ * @brief Get the block hash.
+ *
+ * @param[in] header    Non-null header
+ * @return              Block hash. Must be destroyed btck_block_hash_destroy().
+ */
+BITCOINKERNEL_API btck_BlockHash* BITCOINKERNEL_WARN_UNUSED_RESULT btck_block_header_get_hash(
+    const btck_BlockHeader* header) BITCOINKERNEL_ARG_NONNULL(1);
+
+/**
+ * @brief Get the previous block hash from block header.
+ *
+ * @param[in] header    Non-null header
+ * @return              Previous block hash (view, do not destroy)
+ */
+BITCOINKERNEL_API const btck_BlockHash* BITCOINKERNEL_WARN_UNUSED_RESULT btck_block_header_get_prev_hash(
+    const btck_BlockHeader* header) BITCOINKERNEL_ARG_NONNULL(1);
```

Comment:
We have usually documented the non-owned pointers this way: `The returned X is unowned and only valid for the lifetime of Y.`
```suggestion
/**
 * @brief Get the previous block hash from block header. The returned hash is unowned and only valid for the lifetime of the block header.
 *
 * @param[in] header    Non-null header
 * @return              Previous block hash
 */
BITCOINKERNEL_API const btck_BlockHash* BITCOINKERNEL_WARN_UNUSED_RESULT btck_block_header_get_prev_hash(
    const btck_BlockHeader* header) BITCOINKERNEL_ARG_NONNULL(1);
```